### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by jonathan-caryl**
I saw your library mentioned in Android Weekly. Thought I'd check it out, and I made a few tweaks. 

Hope the PR's ok.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,6 +19,14 @@ allprojects {
     }
 }
 
+ext {
+    compileSdkVersion = 25
+    buildToolsVersion = '25.0.2'
+    minSdkVersion = 16
+    targetSdkVersion = compileSdkVersion
```
Comment (yashasvigirdhar) [inline build.gradle]: `targetSdkVersion` should be separately provided a value. In this case, it happens to be equal to `compileSdkVersion` but that won't be the case always and both shouldn't be dependent on each other

File: build.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:2.3.0'
```
Comment (yashasvigirdhar) [inline build.gradle]: 1) Changing the android gradle plugin version should not be a part of this change. A commit should only target one change at a time, so that if something breaks, we know what caused that.

File: build.gradle [hunk 3]
```diff
@@ -19,6 +19,14 @@ allprojects {
     }
 }
 
+ext {
+    compileSdkVersion = 25
+    buildToolsVersion = '25.0.2'
+    minSdkVersion = 16
+    targetSdkVersion = compileSdkVersion
+
+    supportLibVersion = '25.3.0'
```
Comment (yashasvigirdhar) [inline build.gradle]: Similar to the above comment, you are updating the support lib version here, which should be done in separate commit. This commit should only be a refactor diff, putting out some of the variables in top level gradle file

**Comment by jonathan-caryl** [comment]
I've opened a few other PRs to split things up. Once they've gone in I'll rework this one to do the remaining changes.

**Comment by Krupen** [comment]
Please contribute towards the Library logic to make it better.
